patch/1.25.3.1/ngx_lua-shared_shdict.patch

```diff
@@ -0,0 +1,2965 @@
+index 4741c722..ed0c3a6c 100644
+--- src/ngx_http_lua_headerfilterby.c
++++ src/ngx_http_lua_headerfilterby.c
+@@ -19,7 +19,6 @@
+ #include "ngx_http_lua_string.h"
+ #include "ngx_http_lua_misc.h"
+ #include "ngx_http_lua_consts.h"
+-#include "ngx_http_lua_shdict.h"
+ 
+ 
+ static ngx_http_output_header_filter_pt ngx_http_next_header_filter;
+diff --git src/ngx_http_lua_initby.c src/ngx_http_lua_initby.c
+index e8941da6..4e8d898f 100644
+--- src/ngx_http_lua_initby.c
++++ src/ngx_http_lua_initby.c
+@@ -14,13 +14,13 @@
+ 
+ 
+ ngx_int_t
+-ngx_http_lua_init_by_inline(ngx_log_t *log, ngx_http_lua_main_conf_t *lmcf,
++ngx_http_lua_init_by_inline(ngx_log_t *log, ngx_str_t init_src,
+     lua_State *L)
+ {
+     int         status;
+ 
+-    status = luaL_loadbuffer(L, (char *) lmcf->init_src.data,
+-                             lmcf->init_src.len, "=init_by_lua")
++    status = luaL_loadbuffer(L, (char *) init_src.data, init_src.len,
++                             "=init_by_lua")
+              || ngx_http_lua_do_call(log, L);
+ 
+     return ngx_http_lua_report(log, L, status, "init_by_lua");
+@@ -28,12 +28,12 @@ ngx_http_lua_init_by_inline(ngx_log_t *log, ngx_http_lua_main_conf_t *lmcf,
+ 
+ 
+ ngx_int_t
+-ngx_http_lua_init_by_file(ngx_log_t *log, ngx_http_lua_main_conf_t *lmcf,
++ngx_http_lua_init_by_file(ngx_log_t *log, ngx_str_t init_src,
+     lua_State *L)
+ {
+     int         status;
+ 
+-    status = luaL_loadfile(L, (char *) lmcf->init_src.data)
++    status = luaL_loadfile(L, (char *) init_src.data)
+              || ngx_http_lua_do_call(log, L);
+ 
+     return ngx_http_lua_report(log, L, status, "init_by_lua_file");
+diff --git src/ngx_http_lua_initby.h src/ngx_http_lua_initby.h
+index 67ac0b6c..cbd439aa 100644
+--- src/ngx_http_lua_initby.h
++++ src/ngx_http_lua_initby.h
+@@ -12,10 +12,10 @@
+ 
+ 
+ ngx_int_t ngx_http_lua_init_by_inline(ngx_log_t *log,
+-    ngx_http_lua_main_conf_t *lmcf, lua_State *L);
++    ngx_str_t init_src, lua_State *L);
+ 
+ ngx_int_t ngx_http_lua_init_by_file(ngx_log_t *log,
+-    ngx_http_lua_main_conf_t *lmcf, lua_State *L);
++    ngx_str_t init_src, lua_State *L);
+ 
+ 
+ #endif /* _NGX_HTTP_LUA_INITBY_H_INCLUDED_ */
+diff --git src/ngx_http_lua_initworkerby.c src/ngx_http_lua_initworkerby.c
+index 94de796a..bba63c5b 100644
+--- src/ngx_http_lua_initworkerby.c
++++ src/ngx_http_lua_initworkerby.c
+@@ -293,7 +293,8 @@ ngx_http_lua_init_worker(ngx_cycle_t *cycle)
+ 
+     ngx_http_lua_set_req(lmcf->lua, r);
+ 
+-    (void) lmcf->init_worker_handler(cycle->log, lmcf, lmcf->lua);
++    (void) lmcf->init_worker_handler(cycle->log, lmcf->init_worker_src,
++                                     lmcf->lua);
+ 
+     ngx_destroy_pool(c->pool);
+     return NGX_OK;
+@@ -314,12 +315,12 @@ failed:
+ 
+ ngx_int_t
+ ngx_http_lua_init_worker_by_inline(ngx_log_t *log,
+-    ngx_http_lua_main_conf_t *lmcf, lua_State *L)
++    ngx_str_t init_worker_src, lua_State *L)
+ {
```

Comment:
The implementation has been change in ngx_lua-0.10.26. So the patch will fail to apply